index.js

```diff
@@ -36,9 +36,29 @@ module.exports = function (bot, options) {
     res.sendFile(path.join(__dirname, 'public', 'index.html'))
   })
 
+  function getName (item) {
+    let message = item.displayName
+    if (item.nbt == null) return message
+  
+    const nbt = require('prismarine-nbt')
+    const ChatMessage = require('prismarine-chat')('1.12.2')
+  
+    const data = nbt.simplify(item.nbt)
+    const display = data.display
+    if (display == null) return message
+  
+    const name = display.Name
+    if (name == null) return message
+    message = new ChatMessage(JSON.parse(name)).toString()
+    return message
+  }
+
   io.on('connection', (socket) => {
     // Add item textures
     const items = bot.inventory.itemsRange(0, bot.inventory.inventoryEnd)
+    for (const item in items) {
+      items[item].displayName = getName(items[item])
```

Comment:
Note that you should do the very same thing in the `update` function, otherwise new item/window updates wont appear with the correct displayName